lib/web_ui/lib/src/engine/font_fallbacks.dart

```diff
@@ -257,56 +265,49 @@ class FontFallbackManager {
+      // If the list of best fonts are all CJK fonts, choose the best one based
+      // on user preferred language. Otherwise just choose the first font.
       if (bestFonts.every((NotoFont font) =>
-          font == _notoSansSC ||
-          font == _notoSansTC ||
-          font == _notoSansHK ||
-          font == _notoSansJP ||
-          font == _notoSansKR)) {
-        final String language = domWindow.navigator.language;
-
-        if (language == 'zh-Hans' ||
-            language == 'zh-CN' ||
-            language == 'zh-SG' ||
-            language == 'zh-MY') {
-          if (bestFonts.contains(_notoSansSC)) {
-            bestFont = _notoSansSC;
-          }
-        } else if (language == 'zh-Hant' ||
-            language == 'zh-TW' ||
-            language == 'zh-MO') {
-          if (bestFonts.contains(_notoSansTC)) {
-            bestFont = _notoSansTC;
-          }
-        } else if (language == 'zh-HK') {
-          if (bestFonts.contains(_notoSansHK)) {
-            bestFont = _notoSansHK;
-          }
-        } else if (language == 'ja') {
-          if (bestFonts.contains(_notoSansJP)) {
-            bestFont = _notoSansJP;
-          }
-        } else if (language == 'ko') {
-          if (bestFonts.contains(_notoSansKR)) {
-            bestFont = _notoSansKR;
-          }
-        } else if (bestFonts.contains(_notoSansSC)) {
-          bestFont = _notoSansSC;
+          _isNotoSansSC(font) ||
+          _isNotoSansTC(font) ||
+          _isNotoSansHK(font) ||
+          _isNotoSansJP(font) ||
+          _isNotoSansKR(font))) {
+        if (_language == 'zh-Hans' ||
+            _language == 'zh-CN' ||
+            _language == 'zh-SG' ||
+            _language == 'zh-MY') {
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansSC);
+        } else if (_language == 'zh-Hant' ||
+            _language == 'zh-TW' ||
+            _language == 'zh-MO') {
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansTC);
+        } else if (_language == 'zh-HK') {
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansHK);
+        } else if (_language == 'ja') {
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansJP);
+        } else if (_language == 'ko') {
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansKR);
+        } else {
+          // Default to `Noto Sans SC` when the user preferred language is not CJK.
+          bestFontForLanguage = bestFonts.firstWhereOrNull(_isNotoSansSC);
         }
       } else {
         // To be predictable, if there is a tie for best font, choose a font
         // from this list first, then just choose the first font.
         if (bestFonts.contains(_notoSymbols)) {
           bestFont = _notoSymbols;
-        } else if (bestFonts.contains(_notoSansSC)) {
-          bestFont = _notoSansSC;
+        } else {
+          final notoSansSC = bestFonts.firstWhereOrNull(_isNotoSansSC);
```

Comment:
I guess this final catch-all solves all the possible corner cases of CJK fonts, by defaulting to simplified chinese :P

Reply:
I'm adding a test for this (i.e. `lang="zh"`)